File: backend/app/schemas/forecast.py
from typing import List, Dict, Optional
from pydantic import BaseModel

class HorizonForecastDTO(BaseModel):
    horizon_days: int
    horizon_label: str  # "7-Day Projection" | "30-Day Projection" | "90-Day Projection"
    expected_inflow: float
    expected_outflow: float
    net_cash_flow: float
    projected_closing_balance: float
    confidence_score: int  # 0 to 100
    burn_rate_daily: float
    runway_days: int

class DailyForecastPointDTO(BaseModel):
    date: str
    is_projected: bool
    actual_cash: Optional[float] = None
    projected_cash: Optional[float] = None
    projected_inflow: float = 0.0
    projected_outflow: float = 0.0
    upper_bound: Optional[float] = None
    lower_bound: Optional[float] = None

class LiquidityRiskIndicatorDTO(BaseModel):
    id: str
    risk_title: str
    severity: str  # "Low" | "Medium" | "High" | "Critical"
    threshold_metric: str
    current_status: str
    impact: str
    recommendation: str

class ForecastInsightDTO(BaseModel):
    id: str
    category: str  # "Inflow Surge" | "Working Capital" | "Payroll Timing" | "Reserve Release"
    title: str
    detail: str
    impact_amount: float
    actionable_step: str

class CashForecastResponseDTO(BaseModel):
    executive_summary: str
    current_cash_balance: float
    forecast_7d: HorizonForecastDTO
    forecast_30d: HorizonForecastDTO
    forecast_90d: HorizonForecastDTO
    daily_timeline: List[DailyForecastPointDTO]
    risk_indicators: List[LiquidityRiskIndicatorDTO]
    insights: List[ForecastInsightDTO]

File: apps/merchant-portal/src/types/forecast.ts
export interface HorizonForecastDTO {
  horizon_days: number;
  horizon_label: string;
  expected_inflow: number;
  expected_outflow: number;
  net_cash_flow: number;
  projected_closing_balance: number;
  confidence_score: number;
  burn_rate_daily: number;
  runway_days: number;
}

export interface DailyForecastPointDTO {
  date: string;
  is_projected: boolean;
  actual_cash?: number | null;
  projected_cash?: number | null;
  projected_inflow: number;
  projected_outflow: number;
  upper_bound?: number | null;
  lower_bound?: number | null;
}

export interface LiquidityRiskIndicatorDTO {
  id: string;
  risk_title: string;
  severity: 'Low' | 'Medium' | 'High' | 'Critical';
  threshold_metric: string;
  current_status: string;
  impact: string;
  recommendation: string;
}

export interface ForecastInsightDTO {
  id: string;
  category: string;
  title: string;
  detail: string;
  impact_amount: number;
  actionable_step: string;
}

export interface CashForecastResponseDTO {
  executive_summary: string;
  current_cash_balance: number;
  forecast_7d: HorizonForecastDTO;
  forecast_30d: HorizonForecastDTO;
  forecast_90d: HorizonForecastDTO;
  daily_timeline: DailyForecastPointDTO[];
  risk_indicators: LiquidityRiskIndicatorDTO[];
  insights: ForecastInsightDTO[];
}
